plugins/vbe/fu-vbe-simple-device.c

```diff
@@ -0,0 +1,901 @@
+ *
+ * @parent_instance: FuVbeDevice parent device
+ * @storage: Storage device name (e.g. "mmc1")
+ * @devname: Device name (e.g. /dev/mmcblk1)
+ * @area_start: Start offset of area for firmware
+ * @area_size: Size of firmware area
+ * @skip_offset: This allows an initial part of the image to be skipped when
+ * writing. This means that the first part of the image is ignored, with just
+ * the latter part being written. For example, if this is 0x200 then the first
+ * 512 bytes of the image (which must be present in the image) are skipped and
+ * the bytes after that are written to the store offset.
+ * @fd: File descriptor, if the device is open
+ * @vbe_fname: Filename of the VBE state file
+ * @state: State of this update method
+ */
+struct _FuVbeSimpleDevice {
+	FuVbeDevice parent_instance;
+	const gchar *storage;
+	gchar *devname;
+	off_t area_start;
+	off_t area_size;
+	gint skip_offset;
+	gint fd;
+	gchar *vbe_fname;
+	struct vbe_simple_state state;
+};
+
+G_DEFINE_TYPE(FuVbeSimpleDevice, fu_vbe_simple_device, FU_TYPE_VBE_DEVICE)
+
+/**
+ * trailing_strtoln_end() - Get the number suffix at the end of a string
+ *
+ * Decodes a string like "abc123" into its two parts so that the base string
+ * "abc" and the trailing number "123" can be obtained.
+ *
+ * For this example, with end == NULL, the function returns 123 and *endp
+ * returns a pointer to the '1' character.
+ *
+ * @str: String to parse
+ * @end: End of string to parse, or NULL to parse the whole string
+ * @endp: Returns a pointer to the first character of the numeric suffix. If
+ *	no number is found, this is @end if not NULL, else its points to the
+ *	string's nul terminator
+ * Returns: the numeric value of the suffix, or -1 if none found
+ */
+static gint
+trailing_strtoln_end(const gchar *str, const gchar *end, gchar const **endp)
+{
+	const gchar *p;
+
+	if (!end)
+		end = str + strlen(str);
+	p = end - 1;
+	if (p > str && isdigit(*p)) {
+		do {
+			if (!isdigit(p[-1])) {
+				if (endp)
+					*endp = p;
+				return atoi(p);
+			}
+		} while (--p > str);
+	}
+	if (endp)
+		*endp = end;
+
+	return -1;
+}
+
+/**
+ * fdt_get_u32() - Get a 32-bit integer value from the device tree
+ *
+ * @fdt: Device tree to read from
+ * @node: Node offset to read from
+ * @prop_name: Name of property to read
+ * Returns: integer value, if found and of the correct size, else -1
+ */
+static long
+fdt_get_u32(const gchar *fdt, gint node, const gchar *prop_name)
+{
+	const fdt32_t *val;
+	gint len;
+
+	val = fdt_getprop(fdt, node, prop_name, &len);
+	if (!val || len != sizeof(fdt32_t))
+		return -1;
+
+	return fdt32_to_cpu(*val);
+}
+
+/**
+ * fdt_get_u64() - Get a 64-bit integer value from the device tree
+ *
+ * @fdt: Device tree to read from
+ * @node: Node offset to read from
+ * @prop_name: Name of property to read
+ * Returns: integer value, if found and of the correct size, else -1
+ */
+static long
+fdt_get_u64(const gchar *fdt, gint node, const gchar *prop_name)
```

Comment:
prefix with `fu_vbe_`?  Or if there is reason not to; maybe they should go into your "library" source file.

Reply:
Moved. done